2022-04-22T21:36:31.4410480Z TRACE:: Config.Get(): Configuration key not found. Key: data-directory - Using default value: ../../../Data/
2022-04-22T21:36:31.4449763Z TRACE:: Python for .NET Assembly: Python.Runtime, Version=2.0.11.0, Culture=neutral, PublicKeyToken=5000fea6cba702dd
2022-04-22T21:36:31.4452624Z TRACE:: Config.Get(): Configuration key not found. Key: version-id - Using default value: 
2022-04-22T21:36:31.4472202Z TRACE:: Config.Get(): Configuration key not found. Key: cache-location - Using default value: /Lean/Data
2022-04-22T21:36:31.4479756Z TRACE:: Engine.Main(): LEAN ALGORITHMIC TRADING ENGINE v2.5.0.0 Mode: DEBUG (64bit) Host: DESKTOP-0LAT4UV
2022-04-22T21:36:31.4653751Z TRACE:: Engine.Main(): Started 9:36 PM
2022-04-22T21:36:31.4684128Z TRACE:: Config.Get(): Configuration key not found. Key: lean-manager-type - Using default value: LocalLeanManager
2022-04-22T21:36:31.5718749Z TRACE:: JobQueue.NextJob(): Selected /LeanCLI/main.py
2022-04-22T21:36:31.6406195Z TRACE:: Config.GetValue(): scheduled-event-leaky-bucket-capacity - Using default value: 120
2022-04-22T21:36:31.6409801Z TRACE:: Config.GetValue(): scheduled-event-leaky-bucket-time-interval-minutes - Using default value: 1440
2022-04-22T21:36:31.6413930Z TRACE:: Config.GetValue(): scheduled-event-leaky-bucket-refill-amount - Using default value: 18
2022-04-22T21:36:31.6502012Z TRACE:: Config.Get(): Configuration key not found. Key: job-organization-id - Using default value: 
2022-04-22T21:36:31.6523714Z TRACE:: Config.Get(): Configuration key not found. Key: data-permission-manager - Using default value: DataPermissionManager
2022-04-22T21:36:31.6699936Z TRACE:: AlgorithmManager.CreateTokenBucket(): Initializing LeakyBucket: Capacity: 120 RefillAmount: 18 TimeInterval: 1440
2022-04-22T21:36:31.6728399Z TRACE:: Config.GetValue(): algorithm-manager-time-loop-maximum - Using default value: 20
2022-04-22T21:36:31.6933868Z TRACE:: TextSubscriptionDataSourceReader.SetCacheSize(): Setting cache size to 71582788 items
2022-04-22T21:36:31.9673241Z TRACE:: Config.GetValue(): algorithm-creation-timeout - Using default value: 90
2022-04-22T21:36:31.9711477Z TRACE:: PythonInitializer.Initialize(): start...
2022-04-22T21:36:32.7463726Z TRACE:: PythonInitializer.Initialize(): ended
2022-04-22T21:36:32.7510222Z TRACE:: AlgorithmPythonWrapper(): Python version 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:25:33) 
[GCC 7.3.0]: Importing python module main
2022-04-22T21:36:33.0269627Z ERROR:: Loader.TryCreatePythonAlgorithm():  System.Exception: AlgorithmPythonWrapper(): SyntaxError : invalid syntax (main.py, line 17) SyntaxError : invalid syntax (main.py, line 17)
   at QuantConnect.AlgorithmFactory.Python.Wrappers.AlgorithmPythonWrapper..ctor(String moduleName) in /LeanCloud/CI.Builder/bin/Debug/src/QuantConnect/Lean/AlgorithmFactory/Python/Wrappers/AlgorithmPythonWrapper.cs:line 146
   at QuantConnect.AlgorithmFactory.Loader.TryCreatePythonAlgorithm(String assemblyPath, IAlgorithm& algorithmInstance, String& errorMessage) in /LeanCloud/CI.Builder/bin/Debug/src/QuantConnect/Lean/AlgorithmFactory/Loader.cs:line 173
2022-04-22T21:36:33.0284238Z ERROR:: Engine.Run():  QuantConnect.Lean.Engine.Setup.AlgorithmSetupException: During the algorithm initialization, the following exception has occurred: Loader.TryCreatePythonAlgorithm(): Unable to import python module /LeanCLI/main.py. AlgorithmPythonWrapper(): SyntaxError : invalid syntax (main.py, line 17) SyntaxError : invalid syntax (main.py, line 17)
   at QuantConnect.Lean.Engine.Setup.BacktestingSetupHandler.CreateAlgorithmInstance(AlgorithmNodePacket algorithmNodePacket, String assemblyPath) in /LeanCloud/CI.Builder/bin/Debug/src/QuantConnect/Lean/Engine/Setup/BacktestingSetupHandler.cs:line 104
   at QuantConnect.Lean.Engine.Engine.Run(AlgorithmNodePacket job, AlgorithmManager manager, String assemblyPath, WorkerThread workerThread) in /LeanCloud/CI.Builder/bin/Debug/src/QuantConnect/Lean/Engine/Engine.cs:line 117
2022-04-22T21:36:33.0293487Z TRACE:: JOB HANDLERS: 
2022-04-22T21:36:33.0296434Z TRACE::          DataFeed:     QuantConnect.Lean.Engine.DataFeeds.FileSystemDataFeed
2022-04-22T21:36:33.0300426Z TRACE::          Setup:        QuantConnect.Lean.Engine.Setup.BacktestingSetupHandler
2022-04-22T21:36:33.0302703Z TRACE::          RealTime:     QuantConnect.Lean.Engine.RealTime.BacktestingRealTimeHandler
2022-04-22T21:36:33.0308277Z TRACE::          Results:      QuantConnect.Lean.Engine.Results.BacktestingResultHandler
2022-04-22T21:36:33.0313873Z TRACE::          Transactions: QuantConnect.Lean.Engine.TransactionHandlers.BacktestingTransactionHandler
2022-04-22T21:36:33.0317859Z TRACE::          Alpha:        QuantConnect.Lean.Engine.Alphas.DefaultAlphaHandler
2022-04-22T21:36:33.0321449Z TRACE::          ObjectStore:  QuantConnect.Lean.Engine.Storage.LocalObjectStore
2022-04-22T21:36:33.0331105Z TRACE:: DefaultAlphaHandler.Exit(): Exiting...
2022-04-22T21:36:33.0342006Z TRACE:: DefaultAlphaHandler.Exit(): Ended
2022-04-22T21:36:33.0347490Z TRACE:: BacktestingResultHandler.Exit(): starting...
2022-04-22T21:36:33.0364281Z TRACE:: BacktestingResultHandler.Exit(): Saving logs...
2022-04-22T21:36:33.0433426Z TRACE:: StopSafely(): waiting for 'Result Thread' thread to stop...
2022-04-22T21:36:33.0449403Z ERROR:: Algorithm.Initialize() Error: During the algorithm initialization, the following exception has occurred: Loader.TryCreatePythonAlgorithm(): Unable to import python module /LeanCLI/main.py. AlgorithmPythonWrapper(): SyntaxError : invalid syntax (main.py, line 17) SyntaxError : invalid syntax (main.py, line 17) Stack Trace: During the algorithm initialization, the following exception has occurred: Loader.TryCreatePythonAlgorithm(): Unable to import python module /LeanCLI/main.py. AlgorithmPythonWrapper(): SyntaxError : invalid syntax (main.py, line 17) SyntaxError : invalid syntax (main.py, line 17)
 During the algorithm initialization, the following exception has occurred: Loader.TryCreatePythonAlgorithm(): Unable to import python module /LeanCLI/main.py. AlgorithmPythonWrapper(): SyntaxError : invalid syntax (main.py, line 17) SyntaxError : invalid syntax (main.py, line 17)
2022-04-22T21:36:33.0454624Z TRACE:: Debug: Your log was successfully created and can be retrieved from: /Results/1217306364-log.txt
2022-04-22T21:36:33.0460178Z TRACE:: BacktestingResultHandler.Run(): Ending Thread...
2022-04-22T21:36:33.1330577Z TRACE:: 
STATISTICS:: Fitness Score 0
STATISTICS:: Kelly Criterion Estimate 0
STATISTICS:: Kelly Criterion Probability Value 0
STATISTICS:: Sortino Ratio 0
STATISTICS:: Return Over Maximum Drawdown 0
STATISTICS:: Portfolio Turnover 0
STATISTICS:: Total Insights Generated 0
STATISTICS:: Total Insights Closed 0
STATISTICS:: Total Insights Analysis Completed 0
STATISTICS:: Long Insight Count 0
STATISTICS:: Short Insight Count 0
STATISTICS:: Long/Short Ratio 100%
STATISTICS:: Estimated Monthly Alpha Value $0
STATISTICS:: Total Accumulated Estimated Alpha Value $0
STATISTICS:: Mean Population Estimated Insight Value $0
STATISTICS:: Mean Population Direction 0%
STATISTICS:: Mean Population Magnitude 0%
STATISTICS:: Rolling Averaged Population Direction 0%
STATISTICS:: Rolling Averaged Population Magnitude 0%
STATISTICS:: OrderListHash d41d8cd98f00b204e9800998ecf8427e
2022-04-22T21:36:33.1335000Z TRACE:: BacktestingResultHandler.SendAnalysisResult(): Processed final packet
2022-04-22T21:36:33.1340381Z TRACE:: Engine.Run(): Disposing of setup handler...
2022-04-22T21:36:33.1343400Z TRACE:: Engine.Main(): Analysis Completed and Results Posted.
2022-04-22T21:36:33.1351202Z TRACE:: Engine.Main(): Packet removed from queue: 1217306364
2022-04-22T21:36:33.1355860Z TRACE:: LeanEngineSystemHandlers.Dispose(): start...
2022-04-22T21:36:33.1360041Z TRACE:: LeanEngineSystemHandlers.Dispose(): Disposed of system handlers.
2022-04-22T21:36:33.1363687Z TRACE:: LeanEngineAlgorithmHandlers.Dispose(): start...
2022-04-22T21:36:33.1370023Z TRACE:: LeanEngineAlgorithmHandlers.Dispose(): Disposed of algorithm handlers.
